architecture/fxa.md

```diff
@@ -0,0 +1,232 @@
+# Integration with Firefox Accounts #
+
+Integration with Firefox Accounts (FxA) uses [OAuth](https://oauth.net/); Lockbox applications (Web Extension and mobile) are OAuth public clients and FxA is the authorization service.  It also functions as the user profile service where that information is needed.
+
+## OAuth Protocol Highlights ##
+
+**NOTE:** The OAuth details herein proscribe details and assumptions for Lockbox applications, defining specifics to the options that OAuth allows.
+
+FxA supports [OAuth 2.0][RFC6749] Authorization Code Grant flow, with some additions:
+
+* [PKCE][PKCE] proof keys, using the `S256` challenge mode
+* Application-scoped encryption keys derived from the user's credentials
+
+The complete interactive sign in flow for an FxA OAuth client is:
+
+1. Send an HTTPS `GET` request via a user agent to the authorization endpoint with query parameters
+2. The user agent interacts with the user to complete the authorization request
+3. Upon success, the FxA authorization endpoint redirects to the client's `redirect_uri`, including results as query parameters
+4. The client requests an exchange of the authorization code for access (and refresh) tokens from the token endpoint using an HTTPS `POST` request
+5. The FxA token endpoint validates the request and (upon success) provides the requested tokens
+6. If user information is needed, the client sends an HTTPS `GET` request to the FxA userinfo endpoint (providing the access token as authorization)
+7. The FxA userinfo endpoint validates the authorization and (upon success) provides the user's profile information
+
+FxA access tokens are valid for 1209600 seconds (two weeks) by default, and can be revoked; FxA refresh tokens never expire and can be revoked.
```

Comment:
Nit to add: Tokens are also reset for PKCE clients when user changes or resets the password